README.md

```diff
@@ -192,205 +141,129 @@ data SwapDatum
+## Future Directions
+Features and considerations for future versions of Cardano-Swaps:
 
-#### If Cardano-Swaps reaches mass adoption, won't TVL on Cardano go down?
+### Offer-Based Swap Addresses
+Currently, swap addresses are unique for every trading pair. This necessitates many distinct Beacon UTxOs (and associated deposits), even if users are offering the same asset in exchange for varying assets. Future versions of Cardano-Swaps will instead utilize "Offer-based" swap addresses, which are unique only to the asset being offered. The desired "exchange" asset will be specified in the datum of every "Swap" UTxO at the address. This will dramatically lower Beacon UTxO deposit fees, as only one such UTxO will be needed per "Offer" asset, per user. This will also improve querying efficiency; instead of having to lookup a specific trading pair, users can lookup all swap addresses offering Token "ABC", and then filter through the desired exchange asset(s).
 
-Yes. Yes it will. TVL is a silly metric. It is a measure of who can be most inefficient with DeFi capital.
+> **Note**
+> This feature will require the introduction of a new Beacon Token.
 
-#### I have an idea for how to improve the protocol. Where should I go?
 
-Feel free to use either the Issue section or the Discussion section on the Cardano-Swaps repo. Any feedback that can improve the protocol is welcome!
+### Redundant Script Executions
+Currently, plutus validator scripts must be executed once for *every* UTxO being consumed from a script address (this is the case for Plutus as a whole, not just Cardano-Swaps). If multiple "Swap" UTxOs from the same address are consumed as inputs, the validator must execute for each of them, even though their spending logic is exactly the same. These executions are "redundant", in a sense, resulting in a (low severity) composability bottleneck. Addressing this would likely require significant changes to Cardano's transaction structure, which is unnecessary at this point in time.  [Present and future discussions with the IOG team](https://github.com/cardano-foundation/CIPs/pull/418) will help determine if this feature is indeed warranted.
 
-#### When Mainnet?
+### Swap Collisions
+Recall the composition example above. What would happen if another user simultaneously submits a transaction using *one* of the same "Swap" UTxOs as an input? Whichever transaction is processed by (most) nodes first will succeed, and the other will fail entirely. Arbitragers/market-makers will be in constant competition with one another to execute the most profitable swaps at any point in time, and must strike a balance between simple swaps and complex composed swaps. The greater the input complexity of a large swap, the higher chance that one of the inputs is spent in the interim, making the whole transaction fail.
```

Comment:
You need to add that the chance of collisions goes down with the number of users. And possible give an example: if there are 10,000 available swaps and only 14 fit in a tx, what are the chances that two txs collide?

Reply:
the chances depend on the attractiveness of a swap UTxO. Attractively priced ones are much more likely to collide. I'll mention its less likely the more swaps there are, but its not a linear relationship

Reply:
Thanks to fluid liquidity though, the attractiveness becomes multi-dimensional. The attractiveness depends on how many paths there are to the target arbitrage opportunity. Perhaps you are willing to buy ADA at a 5% premium if it means you can take advantage of an arbitrage between two illiquid assets that yield a 10% return.

Reply:
Also, since the possible swaps is basically infinite, the number of arbitragers actually looking at the same opportunity matters as well.

Reply:
trying hard not to break the reader's brain explaining this